### PR TITLE
<MultiSelectAutocomplete /> component only allows for fetched suggestions now

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Now the `<MultiSelectAutocomplete />` only has the responsibility of allowing existing fetched suggestions to be added as items to the list. Previously, it allowed for new items to be added that were specific to the tags field in the article editor.

From @aitchiss in https://github.com/forem/forem/pull/16604#discussion_r831935722

> Just an idea, but I think perhaps the `MultiSelectAutocomplete` shouldn't be responsible for this.
> 
> So currently:
> 
> 1. We pass in `fetchSuggestions` which returns only existing matching tags from the search call
> 2. In `MultiSelectAutocomplete` if the results are empty, we add the current search term to the suggestions
> 
> Instead we could:
> 
> * Not do point 2 above
> * In `TagsField` which we use for the editor, we could make sure the `fetchSuggestions` prop we pass in adds the current search term as a suggestion if nothing was found in the search call

## Related Tickets & Documents

- Related Issue #16604
- Closes #

## QA Instructions, Screenshots, Recordings

1. Create a new blog post by navigating to http://localhost:3000/new
2. In the tag field, add the existing tag `java`.
3. A dropdown appears with two options.
4. Select `java` from the dropdown or press `,` or the spacebar to insert it into the tags list.
5. Enter a tag that does not exist, e.g. `asdf`
6. 3. A dropdown appears which includes `asdf` in the list.
7. Select `asdf` from the dropdown or press `,` or the spacebar to insert it into the tags list.

### UI accessibility concerns?

Accessibility of the component should be the same as it currently is.

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

Jimmy Fallon saying "Better, faster, stronger

### CODE REVIEW

File: app/javascript/article-form/components/TagsField.jsx [hunk 1]
```diff
@@ -56,8 +56,8 @@ export const TagsField = ({ onInput, defaultValue, switchHelpContext }) => {
   };
 
   const fetchSuggestions = (searchTerm) =>
-    fetchSearch('tags', { name: searchTerm }).then(
-      (response) => response.result,
+    fetchSearch('tags', { name: searchTerm }).then((response) =>
+      response.result.concat({ name: searchTerm }),
```
Comment (nickytonline) [inline app/javascript/article-form/components/TagsField.jsx:60]: For the tag editor, we allow for the creation of new tags, hence the concat

File: app/javascript/crayons/MultiSelectAutocomplete/__tests__/MultiSelectAutocomplete.test.jsx [hunk 2]
```diff
@@ -123,25 +123,6 @@ describe('<MultiSelectAutocomplete />', () => {
     ).toBeInTheDocument();
   });
 
-  it('displays search term as an option if no suggestions', async () => {
```
Comment (nickytonline) [inline app/javascript/crayons/MultiSelectAutocomplete/__tests__/MultiSelectAutocomplete.test.jsx:126]: This test is no longer required as the functionality of the `<MultiSelectAutocomplete />` has changed

**Comment by nickytonline** [comment]
@aitchiss. I've fixed all the E2E tests, but they seem unstable. Most of the time they pass, but on multiple runs random tests fail sporadically. I'm not sure what the issue is at the moment but have run out of time. 🙃  

Once this build is green, update #16604 and that feature should be all good except for the backend work to batch tag moderation updates that @Ridhwana will be looking at.

**Comment by msarit** [comment]
@aitchiss @Ridhwana Hey hey! Who's running point on this PR now? 🙏🏾

**Comment by aitchiss** [comment]
Hey @msarit, it's assigned to me, but it was put on the back-burner while we tackled the member index view project. I'm likely to be picking this up again very shortly 🙂

**Comment by aitchiss** [comment]
Just a small update - I was looking into the failing specs, and I think I've spotted an issue that suggests Nick's original plan of passing a prop into the `MultiSelectComponent` is actually a better way forward (my bad!)

The main gist is that relying on the network fetch completing before allowing a typed suggestion to be selected by keyboard/blur causes:

- Kind of an annoying lag as a user typing confidently 😄 
- The functionality where we can _split_ a tag (e.g. changing `onetwo` to `one two` by editing to add a space) no longer works

I'm going to rework this closer to Nick's original approach, passing a prop in instead.

**Comment by aitchiss** [comment]
Closing in favour of https://github.com/forem/forem/pull/17402

**Pull request closed without merging**